source/constructs/api/template/service.py

```diff
@@ -167,10 +171,230 @@ def sync_s3(snapshot_no):
+            if label:
+                sheet.cell(row=row_num, column=9, value=label)
+    workbook.active = 0
+    file_name = f"identify_{key}.xlsx"
+    tmp_file = f"{tempfile.gettempdir()}/{file_name}"
+    report_file = f"{const.IDENTIFY_REPORT}/{file_name}"
+    workbook.save(tmp_file)
+    stats = os.stat(tmp_file)
+    if stats.st_size < 6 * 1024 * 1024:
+        __s3_client.upload_file(tmp_file, admin_bucket_name, report_file)
+    else:
+        concurrent_upload2s3(admin_bucket_name, report_file, tmp_file, __s3_client)
+    os.remove(tmp_file)
+    method_parameters = {'Bucket': admin_bucket_name, 'Key': report_file}
+    pre_url = __s3_client.generate_presigned_url(
+        ClientMethod="get_object",
+        Params=method_parameters,
+        ExpiresIn=60
+    )
+    return pre_url
+
+def __convert_prop_list_2_mapping(list):
+    mapping = {}
+    for item in list:
+        mapping[f"{item[0]}-{item[1]}"] = item[2]
+    return mapping
+
+def delete_report(key):
+    __s3_client.delete_object(Bucket=admin_bucket_name, Key=f"{const.BATCH_CREATE_IDENTIFIER_REPORT_PATH}/{key}.xlsx")
+
+
+def query_batch_status(filename: str):
+    success, warning, failed = 0, 0, 0
+    file_key = f"{const.BATCH_CREATE_IDENTIFIER_REPORT_PATH}/{filename}.xlsx"
+    response = __s3_client.list_objects_v2(Bucket=admin_bucket_name, Prefix=const.BATCH_CREATE_IDENTIFIER_REPORT_PATH)
+    for obj in response.get('Contents', []):
+        if obj['Key'] == file_key:
+            response = __s3_client.get_object(Bucket=admin_bucket_name, Key=file_key)
+            excel_bytes = response['Body'].read()
+            workbook = openpyxl.load_workbook(BytesIO(excel_bytes))
+            try:
+                sheet = workbook[const.BATCH_SHEET]
+            except KeyError:
+                raise BizException(MessageEnum.SOURCE_BATCH_SHEET_NOT_FOUND.get_code(),
+                                   MessageEnum.SOURCE_BATCH_SHEET_NOT_FOUND.get_msg())
+            for _, row in enumerate(sheet.iter_rows(values_only=True, min_row=3)):
+                if row[9] == "FAILED":
+                    failed += 1
+                if row[9] == "SUCCESSED":
+                    success += 1
+                if row[9] == "WARNING":
+                    warning += 1
+            return {"success": success, "warning": warning, "failed": failed}
+    return 0
+
+def download_batch_file(filename: str):
+    key = f'{const.BATCH_CREATE_IDENTIFIER_REPORT_PATH}/{filename}.xlsx'
+    if filename.startswith("identifier-template-zh"):
+        key = const.BATCH_CREATE_IDENTIFIER_TEMPLATE_PATH_CN
+    if filename.startswith("identifier-template-en"):
+        key = const.BATCH_CREATE_IDENTIFIER_TEMPLATE_PATH_EN
+    url = __s3_client.generate_presigned_url(
+        ClientMethod="get_object",
+        Params={'Bucket': admin_bucket_name, 'Key': key},
+        ExpiresIn=60
+    )
+    return url
+
+def batch_create(file: UploadFile = File(...)):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this function is 32. By comparison, 99% of the functions in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the function has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this function or break it into multiple functions. For example, consider extracting the code block on lines 339-349 into a separate function.